openapi/shopify.admin/openapi.yml

```
@@ -3338,7 +3338,9 @@ components:
           type: integer
           description: The unique identifier for the refund that the order adjustment is associated with.
         amount:
-          type: integer
+          oneOf:
+            - type: integer
+            - type: string
```

Just notice that the type you want to generate is nullable too. in such case you need to make the field nullable too. 

```suggestion
          oneOf:
            - type: integer
          	  nullable: true
            - type: string
          	  nullable: true
```